test/testing/test_services.py

```diff
@@ -31,6 +35,23 @@ def reset_mock():
     handle_event.reset_mock()
 
 
+@pytest.yield_fixture
+def spawn_thread():
+
+    threads = []
+
+    def spawn(fn, *args):
+        threads.append(eventlet.spawn(fn, *args))
+
+    yield spawn
+
+    for gt in threads:
+        try:
+            gt.kill()
+        except:
```

Comment:
bare `except:` are almost always mistakes (or laziness). the gotcha is that it catches stuff that doesn't inherit from `Exception` (e.g. `KeyboardInterrupt` as you note)

i think `GreenletException` also doesn't inherit from `Exception`, so worth checking what we want to do here, and either catch it explicitly, or, if you really want to catch everything, catch `BaseException` to indicate it's on purpose (and comment as to why)